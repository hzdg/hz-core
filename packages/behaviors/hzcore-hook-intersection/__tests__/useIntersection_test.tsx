/* eslint-env jest, browser */
import React from 'react';
import {render} from 'react-testing-library';
import useIntersection from '../src';

test('useIntersection gets the initial intersects', async () => {
  let intersects;
  const IntersectionUser = (): JSX.Element => {
    let ref;
    [intersects, ref] = useIntersection<HTMLDivElement>();
    return <div ref={ref} data-testid="intersects" />;
  };
  render(<IntersectionUser />);
  expect(intersects).toBe(false);
});

test('useIntersection uses an existing ref', async () => {
  let intersects;
  const IntersectionUser = (): JSX.Element => {
    const ref = React.useRef(null);
    intersects = useIntersection<HTMLDivElement>(ref);
    return <div ref={ref} data-testid="intersects" />;
  };
  render(<IntersectionUser />);
  expect(intersects).toBe(false);
});

test('useIntersection uses a handler', async () => {
  let ref;
  const IntersectionUser = (): JSX.Element => {
    ref = useIntersection<HTMLDivElement>(() => {});
    return <div ref={ref} data-testid="intersects" />;
  };
  render(<IntersectionUser />);
  expect(ref).toMatchSnapshot();
});

test('useIntersection uses an existing ref and a handler', async () => {
  let intersects;
  const IntersectionUser = (): JSX.Element => {
    const ref = React.useRef(null);
    intersects = useIntersection<HTMLDivElement>(ref, () => {});
    return <div ref={ref} data-testid="intersects" />;
  };
  render(<IntersectionUser />);
  expect(intersects).toBeUndefined();
});
