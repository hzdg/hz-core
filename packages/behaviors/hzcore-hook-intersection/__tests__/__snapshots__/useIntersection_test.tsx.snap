// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useIntersection uses a handler 1`] = `
Object {
  "current": <div
    data-testid="intersects"
  />,
}
`;
